feat: support VMware workstation provider

Gexinyu2333 opened the pull request:
When a provider adds resources in the form of VMware on the providers interface. As shown below:
![b596834870087c86e5de667c2f823e6](https://github.com/user-attachments/assets/8dc126a2-947b-460f-a371-e1e1a3ca5163)

The Client ID is a virtual machine ID of VMware Workstation Pro belonging to the provider; Client secret is the credential for VMware Workstation Pro REST service, in the form of {username}:{password}
If the VMware Workstation Pro REST service of the provider host is enabled, all virtual machines information in the provider's VMware Workstation Pro will be displayed on the machines page. As shown in the following figure:
![e53f78b1ea8c4d359878b1752c2b653](https://github.com/user-attachments/assets/9014469d-d8a0-43fa-8d03-22aa253f5d6f)

hsluoyz commented on object/machine.go:
```diff
@@ -95,8 +95,7 @@ func GetMachines(owner string) ([]*Machine, error) {
 		}
-
-		clientMachines, err2 := client.GetMachines()
+		clientMachines, err2 := (*client).GetMachines()
```
Rebase your code onto the master HEAD

hsluoyz commented on object/machine.go:
```diff
@@ -135,14 +134,13 @@ func getMachine(owner string, name string) (*Machine, error) {
 		}
 
-		clientMachine, err2 := client.GetMachine(name)
+		clientMachine, err2 := (*client).GetMachine(name)
```
Rebase your code onto the master HEAD

hsluoyz commented on service/aliyun.go:
```diff
@@ -50,17 +50,17 @@ type MachineAliyunClient struct {
 }
 
-func NewMachineAliyunClient(accessKeyId string, accessKeySecret string, region string) (*MachineAliyunClient, error) {
+func NewMachineAliyunClient(accessKeyId string, accessKeySecret string, region string) (MachineAliyunClient, error) {
```
Rebase your code onto the master HEAD

hsluoyz commented on service/util.go:
```diff
@@ -30,3 +34,15 @@ func getLocalTimestamp(input string) string {
 	return res
 }
+
+func ExtractFileName(path string) string {
```
Go std lib should have this func

hsluoyz commented on service/vmware.go:
```diff
@@ -0,0 +1,129 @@
+// you may not use this file except in compliance with the License.
```
Not correct Apache header

hsluoyz commented on service/vmware.go:
```diff
@@ -0,0 +1,129 @@
+	if err != nil {
+		return nil, err
+	}
+	if len(vmList) == 0 {
```
Remove this if

hsluoyz commented on service/vmware.go:
```diff
@@ -0,0 +1,129 @@
+	for _, vm := range vmList {
+		client.accessKeyId = vm.ID
+		name := ExtractFileName(vm.Path)
+		machine, err := client.GetMachine(name)
```
Use Go routine to speed up

hsluoyz commented on service/vmware.go:
```diff
@@ -0,0 +1,129 @@
+
+func (client MachineVMwareClient) GetMachines() ([]*Machine, error) {
+	machines := []*Machine{}
+	url := "http://localhost:8697/api/vms"
```
`http://localhost:8697` should be a configurable field in frontend. Skip this if you don't know how to do it

hsluoyz reviewed (changes requested):
Demo video?

hsluoyz commented on service/vmware.go:
```diff
@@ -0,0 +1,129 @@
+	"net/http"
+)
+
+type MachineVMwareClient struct {
```
Use `Vmware` for all variable naming

Gexinyu2333 committed:
Update machine.go

feat: #139 support showing VMware virtual machines in machine list and edit pages

Gexinyu2333 committed:
Update aliyun.go

feat: #139 support showing VMware virtual machines in machine list and edit pages

Gexinyu2333 committed:
Update machine.go

feat: #139 support showing VMware virtual machines in machine list and edit pages

Gexinyu2333 committed:
Create vmware

feat: #139 support showing VMware virtual machines in machine list and edit pages

Gexinyu2333 committed:
Update machine.go

feat: #139 support showing VMware virtual machines in machine list and edit pages

Gexinyu2333 committed:
feat: #139 support showing VMware virtual machines in machine list and edit pages

Rename vmware to vmware.go
